From 6f451bd573162c216a621b33dcadc635dbb1c52a Mon Sep 17 00:00:00 2001
From: Karthik Nayak <Karthik.188@gmail.com>
Date: Wed, 27 Apr 2016 15:19:45 +0530
Subject: [PATCH] MPU: flip the roll axis

Signed-off-by: Karthik Nayak <Karthik.188@gmail.com>
---
 MPU6050_6Axis_MotionApps20.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MPU6050_6Axis_MotionApps20.h b/MPU6050_6Axis_MotionApps20.h
index 9e2c478..6b70c73 100644
--- a/MPU6050_6Axis_MotionApps20.h
+++ b/MPU6050_6Axis_MotionApps20.h
@@ -666,7 +666,7 @@ uint8_t MPU6050::dmpGetYawPitchRoll(float *data, Quaternion *q, VectorFloat *gra
     // pitch: (nose up/down, about X axis)
     data[1] = atan(gravity -> y / sqrt(gravity -> x*gravity -> x + gravity -> z*gravity -> z));
     // roll: (tilt left/right, about Y axis)
-    data[2] = atan(gravity -> x / sqrt(gravity -> y*gravity -> y + gravity -> z*gravity -> z));
+    data[2] = -atan(gravity -> x / sqrt(gravity -> y*gravity -> y + gravity -> z*gravity -> z));
     return 0;
 }
 
-- 
2.8.0

